wp_org_api/src/plugin_directory.rs

```diff
@@ -0,0 +1,542 @@
+            slug,
+            result.err()
+        );
+
+        assert!(!result.unwrap().tags.is_empty());
+    }
+
+    #[rstest]
+    fn test_property_empty_versions(#[values("mos-faqs", "adjustly-collapse")] slug: &str) {
+        let result = parse_plugin(slug);
+        assert!(
+            result.is_ok(),
+            "Failed to parse plugin {:?}: {:?}",
+            slug,
+            result.err()
+        );
+
+        assert!(result.unwrap().versions.is_empty());
+    }
+
+    #[rstest]
+    fn test_property_nonempty_versions(#[values("abcsubmit", "acf-views")] slug: &str) {
+        let result = parse_plugin(slug);
+        assert!(
+            result.is_ok(),
+            "Failed to parse plugin {:?}: {:?}",
+            slug,
+            result.err()
+        );
+
+        assert!(!result.unwrap().versions.is_empty());
+    }
+
+    #[rstest]
+    #[case(
+        "appmysite",
+        "https://ps.w.org/appmysite/assets/banner-772x250.png?rev=2829272",
+        "https://ps.w.org/appmysite/assets/banner-1544x500.png?rev=2829272"
+    )]
+    #[case(
+        "jetpack",
+        "https://ps.w.org/jetpack/assets/banner-772x250.png?rev=2653649",
+        "https://ps.w.org/jetpack/assets/banner-1544x500.png?rev=2653649"
+    )]
+    #[case(
+        "1-click-migration",
+        "https://ps.w.org/1-click-migration/assets/banner-772x250.png?rev=2333853",
+        ""
+    )]
+    fn test_property_banners(#[case] slug: &str, #[case] low: String, #[case] high: String) {
+        let result = parse_plugin(slug);
+        assert!(
+            result.is_ok(),
+            "Failed to parse plugin {:?}: {:?}",
+            slug,
+            result.err()
+        );
+
+        let expected = Banners { low, high };
+        let banners = result.unwrap().banners;
+        assert_eq!(banners, expected);
+    }
+
+    #[rstest]
+    #[case(
+        "contact-form-7",
+        Some("https://ps.w.org/contact-form-7/assets/icon.svg?rev=2339255"),
+        None,
+        Some("https://ps.w.org/contact-form-7/assets/icon.svg?rev=2339255"),
+        None
+    )]
+    #[case(
+        "adminimize",
+        None,
+        None,
+        None,
+        Some("https://s.w.org/plugins/geopattern-icon/adminimize_000000.svg")
+    )]
+    fn test_property_icons(
+        #[case] slug: &str,
+        #[case] low: Option<&str>,
+        #[case] high: Option<&str>,
+        #[case] svg: Option<&str>,
+        #[case] default: Option<&str>,
+    ) {
+        let result = parse_plugin(slug);
+        assert!(
+            result.is_ok(),
+            "Failed to parse plugin {:?}: {:?}",
+            slug,
+            result.err()
+        );
+
+        let icons = result.unwrap().icons;
+        assert!(icons.is_some());
+
+        let icons = icons.unwrap();
+        assert_eq!(icons.low.as_deref(), low);
+        assert_eq!(icons.high.as_deref(), high);
+        assert_eq!(icons.svg.as_deref(), svg);
+        assert_eq!(icons.default.as_deref(), default);
```

Comment:
I don't think `default` is a [reserved keyword](https://doc.rust-lang.org/book/appendix-01-keywords.html) 😕